app.py

```diff
@@ -9,11 +14,42 @@ def index():
 
 @app.route("/dashboard")
 def dashboard():
-	return render_template("dashboard.html")
+	os.makedirs(UPLOAD_FOLDER, exist_ok=True)
+	files = [file.removeprefix(UPLOAD_FOLDER+'/') for file in glob(UPLOAD_FOLDER+'/*.c')]
+	return render_template("dashboard.html", files=files)
 
-@app.route("/dashboard/result")
-def result():
-	return render_template("result.html")
+@app.route('/upload', methods=['POST'])
+def upload_files():
+    if 'files' not in request.files:
+        flash('No file part in the request')
+        return redirect(request.url)
+
+    files = request.files.getlist('files')
+    if not files:
+        flash('No files selected')
+        return redirect(request.url)
+
+    for file in files:
+        if file and allowed_file(file.filename):
+            file.save(os.path.join(UPLOAD_FOLDER, file.filename))
```

Comment:
last time i heard that this website will only be used internally, but just to let you guys know that this is a security vulnerability right here

since you didnt sanitize the `file.filename`, if the user upload a certain file with the filename `/../../../../../../etc/passwd` it will effectively override the system `/etc/passwd` which is a path traversal vulnerability and could lead to system wide takeover

please always sanitize user inputs wherever you need to, so for this one please sanitize the `file.filename` first, and it will be a good exercise for you to figure out how to sanitize it 👌

Reply:
Alright sir, we'll do the next commit the sanitized one of the `file.filename` input to be more secure later...
We could do the parsing and having a custom lexer (or just use regex) to solve this particular problem!

Reply:
no need to do any parsing or custom lexer actually 😅 , you guys just need to encode some if not all of the possibly harmful characters in the string 👍